eXo/Sources/Controllers/HomePage/HomePageViewController.swift

```diff
@@ -285,12 +287,29 @@ final class HomePageViewController: eXoWebBaseController, WKNavigationDelegate,
         if  (request.url?.path.contains("/jitsiweb/"))! && (request.url?.absoluteString.range(of:"?jwt=") != nil){
             // launch the call.
             print("launching the call...")
-            // launch the call using the app.
-            if let jitsiUrl = request.url, (jitsiUrl.absoluteString.range(of: "org.jitsi.meet:") != nil) {
-                if UIApplication.shared.canOpenURL(jitsiUrl){
-                    UIApplication.shared.open(jitsiUrl)
-                } else {
-                    //Show alert message because the user doesn't have Jitsi App installed.
+            // Check first if the user has the jitsi app installed.
+            if isJitsiAppInstalled() {
+                // launch the call using the app.
+                if let jitsiUrl = request.url {
+                    let components = URLComponents(url: jitsiUrl,resolvingAgainstBaseURL: false)!
+                    // URL query parameters as a dictionary
+                    var newUrlComponents = URLComponents()
+                    newUrlComponents.scheme = "org.jitsi.meet"
+                    newUrlComponents.host = components.host
+                    newUrlComponents.path = components.path
+                    newUrlComponents.queryItems = components.queryItems
+                    if let url = newUrlComponents.url {
+                        if UIApplication.shared.canOpenURL(url){
+                            UIApplication.shared.open(url)
+                            if let wv = popupWebView {
+                                webViewDidClose(wv)
+                            }
+                        }
+                    }
+                }
+            }else{
+                if let jitsiUrl = request.url, (jitsiUrl.absoluteString.range(of: "org.jitsi.meet:") != nil) {
+                    //User has tapped on use jitsi App button, so show alert message to the user.
                     showAlertMessage(title: "Jitsi Alert", msg: "Jitsi meet is not on your mobile, install it before.", action: .defaultAction)
```

Comment:
Does this mean we won't let the user use the Web browser if the Jitsi app is not installed ?

Reply:
No, this line of code means that as I mentioned in the comment will be triggered just if the user presses "Use jitsi App" button, else it will do nothing and the code will continue running normally.